common/md5.h

```diff
@@ -49,6 +51,7 @@ class String;
  * @return true on success, false if an error occurred
  */
 bool computeStreamMD5(ReadStream &stream, uint8 digest[16], uint32 length = 0);
+bool computeStreamMD5(ReadStream &stream, uint8 digest[16], uint32 length, ProgressUpdateCallback progressUpdateCallback );
```

Comment:
No, please avoid this kind of unnecessary overloading. Instead, make it as an optional parameter.

You brought way too much copy/paste.